File: src/main/java/tk/ardentbot/commands/botAdministration/Request.java
package tk.ardentbot.commands.botAdministration;

import net.dv8tion.jda.core.entities.*;
import tk.ardentbot.core.executor.Ratelimitable;
import tk.ardentbot.utils.discord.GuildUtils;

import static tk.ardentbot.main.Ardent.botLogsShard;

public class Request extends Ratelimitable {
    public Request(CommandSettings commandSettings) {
        super(commandSettings);
    }

    @Override
    public void noArgs(Guild guild, MessageChannel channel, User user, Message message, String[] args) throws Exception {
        String prefix = GuildUtils.getPrefix(guild);
        if (args.length == 1) {
            sendTranslatedMessage(getTranslation("request", language, "requesthelp").getTranslation().replace("{0}",
                    prefix + args[0]), channel, user);
        }
        else {
            String request = message.getRawContent().replace(prefix + args[0] + " ", "");
            TextChannel ideasChannel = botLogsShard.jda.getTextChannelById("262810786186002432");
            ideasChannel.sendMessage("**Request** by " + user.getName() + "#" + user.getDiscriminator() + " (" +
                    user.getId() + "): " + request).queue();
            sendRetrievedTranslation(channel, "request", language, "successfullyrequested", user);
        }
    }

    @Override
    public void setupSubcommands() {
    }
}

File: src/main/java/tk/ardentbot/commands/botAdministration/Request.kt
package tk.ardentbot.commands.botAdministration

import net.dv8tion.jda.core.entities.Guild
import net.dv8tion.jda.core.entities.Message
import net.dv8tion.jda.core.entities.MessageChannel
import net.dv8tion.jda.core.entities.User
import tk.ardentbot.core.executor.Ratelimitable
import tk.ardentbot.main.Ardent.botLogsShard
import tk.ardentbot.utils.discord.GuildUtils


class Request(commandSettings: CommandSettings) : Ratelimitable(commandSettings) {
    override fun noArgs(guild: Guild, channel: MessageChannel?, user: User, message: Message, args: Array<out String>?) {
        if (args!!.size == 1) {
            sendTranslatedMessage("Request a feature by typing /request and then your request", channel, user)
            return
        }
        val prefix = GuildUtils.getPrefix(guild)
        val request = message.rawContent.replace(prefix + args[0] + " ", "")
        val ideasChannel = botLogsShard.jda.getTextChannelById("262810786186002432")
        ideasChannel.sendMessage("Request: " + user.name + "#" + user.discriminator + " (" +
                guild.name + "): " + request).queue()
        sendTranslatedMessage("Sent your request to Ardent's owner! He will message you if your suggestion will be implemented", channel, user)
    }

    override fun setupSubcommands() {
    }
}